timelock-corruption-detection/src/main/java/com/palantir/history/models/LearnerAndAcceptorRecords.java

```diff
@@ -31,4 +33,34 @@
 
     @Value.Parameter
     Map<Long, PaxosAcceptorState> acceptorRecords();
+
+    default Optional<PaxosValue> getLearnedValueAtSeqIfExists(long seq) {
+        return Optional.ofNullable(learnerRecords().getOrDefault(seq, null));
+    }
+
+    default Optional<PaxosAcceptorState> getAcceptedValueAtSeqIfExists(long seq) {
+        return Optional.ofNullable(acceptorRecords().getOrDefault(seq, null));
+    }
+
+    default long getMinSequence() {
+        long minSeq = Long.MAX_VALUE;
+        if (!learnerRecords().isEmpty()) {
+            minSeq = Math.min(minSeq, Collections.min(learnerRecords().keySet()));
+        }
+        if (!acceptorRecords().isEmpty()) {
+            minSeq = Math.min(minSeq, Collections.min(acceptorRecords().keySet()));
+        }
+        return minSeq;
```

Comment:
Yeah, I think this function is correct, though this is maybe a bit neater?

```suggestion
        return Stream.of(learnerRecords(), acceptorRecords())
                .map(Map::keySet)
                .flatMap(Set::stream)
                .min(Comparator.naturalOrder())
                .orElse(Long.MAX_VALUE);
```